FirebaseAdmin/FirebaseAdmin/Auth/Hash/Hmac.cs

```diff
@@ -28,30 +27,30 @@ public abstract class Hmac : UserImportHash
         /// Propogates the name to UserImportHash.
         /// </summary>
         /// <param name="hashName">The name of the hashing algorithm.</param>
-        public Hmac(string hashName)
+        internal Hmac(string hashName)
             : base(hashName) { }
 
         /// <summary>
         /// Gets or sets the key for the hash.
         /// </summary>
-        public string Key { get; set; }
+        public byte[] Key { get; set; }
 
         /// <summary>
         /// Verifies that the is key non-empty or null and returns the options dictionary.
         /// </summary>
         /// <returns>
         /// Returns the dictionary containing an entry for the signing key.
         /// </returns>
-        protected override IReadOnlyDictionary<string, object> GetOptions()
+        protected override IReadOnlyDictionary<string, object> GetHashConfiguration()
         {
-            if (string.IsNullOrEmpty(this.Key))
+            if (this.Key == null || this.Key.Length == 0)
             {
                 throw new ArgumentException("key must not be null or empty");
             }
 
             return new Dictionary<string, object>
             {
-                { "signerKey", Encoding.ASCII.GetBytes(this.Key) },
+                { "signerKey", this.Key },
```

Comment:
Shouldn't you have to base64 encode the byte array? I think otherwise json marshalling will break down.